src/js_parser.zig

```diff
@@ -17355,9 +17356,151 @@ fn NewParser_(
         ) []Stmt {
             switch (stmtorexpr) {
                 .stmt => |stmt| {
-                    var stmts = p.allocator.alloc(Stmt, 1) catch unreachable;
-                    stmts[0] = stmt;
-                    return stmts;
+                    var class = stmt.data.s_class.class;
+                    var ctor: ?*E.Function = null;
+
+                    var hasDecorators: bool = class.ts_decorators.len > 0;
+                    for (class.properties) |prop| {
+                        if (prop.ts_decorators.len > 0) {
+                            hasDecorators = true;
+                            break;
+                        }
+                    }
+
+                    if (!hasDecorators) {
+                        var stmts = p.allocator.alloc(Stmt, 1) catch unreachable;
+                        stmts[0] = stmt;
+                        return stmts;
+                    }
+
+                    var stmts = ListManaged(Stmt).init(p.allocator);
+
+                    var instanceMembers = ListManaged(Stmt).init(p.allocator);
```

Comment:
These temporary arrays will make this slower, but correctly counting each item is brittle code. We probably should have an abstraction for this as a pattern - taking an ArrayList(Type), counting elements to insert and then splitting into separate arrays that are appendable. No action necessary for now